sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -58,22 +89,26 @@
 
 import javax.annotation.Nullable;
 
+
 /**
  * Utility implementing the necessary reflection for working with {@link DoFnWithContext}s.
  */
 public abstract class DoFnReflector {
 
+  private static final String FN_DELEGATE_FIELD_NAME = "delegate";
+
   private interface ExtraContextInfo {
-    /**
-     * Create an instance of the given instance using the instance factory.
-     */
-    <InputT, OutputT> Object createInstance(
-        DoFnWithContext.ExtraContextFactory<InputT, OutputT> factory);
 
     /**
      * Create the type token for the given type, filling in the generics.
      */
     <InputT, OutputT> TypeToken<?> tokenFor(TypeToken<InputT> in, TypeToken<OutputT> out);
+
+    /**
+     * Return the {@link MethodDescription} for accessing this context info from the
+     * {@link ExtraContextFactory}.
+     */
+    MethodDescription getMethodDescription() throws NoSuchMethodException, SecurityException;
```

Comment:
Apologies, but I'm nitpicking naming readability because this code is fairly subtle. It seems that `ExtraContextInfo` is essentially a dynamic parameter description, with type. Naming and/or javadoc would give some good hints in this direction.